### ISSUE TITLE
Update OPAL to latest version?

### ISSUE THREAD

**Issue by amottier** [fasten-project/fasten#314]
## Is your feature request related to a problem? Please describe.
This is not actually related to a problem, just that we are currently using a SNAPSHOT version (`3.0.0-SNAPSHOT`) that has been superseded by a more recent stable version (`4.0.0`).

## Describe the solution you'd like
Upgrading to version 4.0.0 available from Maven Central https://search.maven.org/search?q=de.opal-project

## Additional context
According to the [release notes](https://github.com/opalj/opal/blob/develop/Changes.md#400---released-may-7th-2021) version 4.0.0 brings full support for Java 11 up to 16. That should help to fix #173.

**Comment by proksch** [comment]
I have updated OPAL to the latest version in #343.

**Comment by proksch** [comment]
Our current tests all run successfully and I have conducted am additional, preliminary test with a handful of randomly picked Maven coordinates, in which I compared the results of both versions... the results seems to be mostly stable (+- one digit number of edges in some cases), so I do not expect any larger changes.

In addition to all the changes that OPAL brings under the hood, for us, the main difference is that we could observe a weird inconsistency in v3 that forced us to instantiate OPAL in a very specific way to achieve a stable number of edges. This is no longer necessary and we could cut-down the OPAL setup to essentially three lines of code and three config parameters that go beyond the default config.

What is still missing is a test to check if the various cases, in which OPAL crashes during the call graph generation, still appear in the new version. I hope that the new version fixes some of these, but I do not know yet... @ashkboos will look into this in the next days.
